sky/shell/platform/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -62,53 +71,64 @@ - (instancetype)initWithBridge:(sky::shell::AccessibilityBridge*)bridge
   return self;
 }
 
-#pragma mark - Semantics node methods
-
-- (void)update:(const semantics::SemanticsNodePtr&)mojoNode {
-  DCHECK(_uid == mojoNode->id);
-
-  if (!mojoNode->flags.is_null()) {
-    _flags = mojoNode->flags.Pass();
-  }
-
-  if (!mojoNode->strings.is_null()) {
-    _strings = mojoNode->strings.Pass();
-  }
-
-  if (!mojoNode->geometry.is_null()) {
-    _geometry = mojoNode->geometry;
-  }
-
-  if (!mojoNode->children.is_null()) {
-    // Mark existing children as orphans
-    NSArray* oldChildren = _children;
-    for (AccessibilityNode* child in oldChildren) {
-      DCHECK(child->_parent != nil);
-      child->_parent = nil;
-    }
-
-    // Set the new list of children
-    NSMutableArray* children = [[NSMutableArray alloc] init];
-    _children = children;
-    for (const semantics::SemanticsNodePtr& mojoChild : mojoNode->children) {
-      AccessibilityNode* child = _bridge->UpdateNode(mojoChild);
-      child->_parent = self;
-      [children insertObject:child atIndex:0];
-    }
-
-    // Remove those children that are still marked as orphans
-    for (AccessibilityNode* child in oldChildren) {
-      if (child->_parent == nil) {
-        _bridge->RemoveNode(child);
+#pragma mark - Semantic object methods
+
+- (void)updateWith:(const semantics::SemanticsNodePtr&)node {
+  DCHECK(_uid == node->id);
+
+  if (!node->flags.is_null())
+    _flags = node->flags.Pass();
+
+  if (!node->strings.is_null())
+    _strings = node->strings.Pass();
+
+  if (!node->geometry.is_null())
+    _geometry = node->geometry;
+
+  if (!node->actions.is_null()) {
+    _canBeTapped = false;
+    _canBeLongPressed = false;
+    _canBeScrolledHorizontally = false;
+    _canBeScrolledVertically = false;
+    for (semantics::SemanticAction action : node->actions) {
+      switch (action) {
+      case semantics::SemanticAction::TAP:
+        _canBeTapped = true;
+        break;
+      case semantics::SemanticAction::LONG_PRESS:
+        _canBeLongPressed = true;
+        break;
+      case semantics::SemanticAction::SCROLL_LEFT:
```

Comment:
You can coalesce the `SCROLL_LEFT` and `SCROLL_RIGHT` cases.  Same with `SCROLL_UP` and `SCROLL_DOWN`